src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -301,6 +301,11 @@ public function getProductType(ProductId $productId): ProductType
             ));
         }
 
+        if ($result['product_type'] === ProductType::TYPE_UNDEFINED) {
```

Comment:
That's a good initiative to make this method more resilient to absent data in the DB! I was afraid that maybe we were actually relying on the fact that this exception needs to be thrown in some cases but it doesn't seem to be the case.

To validate this new behavior and make sure we don't include some regression later we should enforce this with a test. This could be tested in behat test by creating a Product with an undefined type maybe in an integration test

Reply:
After some investigations, I found those tests : https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Scenario/Product/legacy_product_type.feature
Isn't it enough ? What test is lacking for you ?

Reply:
This test was already present before your fix, and yet a bug was happening when we requested the product's suppliers using the `GetProductSupplierOptions` query Which means this use case was not tested (trying to ftehc the suppliers while the type is undefined)

So our test coverage didn't handle this buggy use case, meaning the use case wasn't present and should be added, and you're right it should probably be tested in the feature you mentioned

Basically, you can add a test that tries and get the suppliers, if you remove your fix it should fail but with the fix it should be fine now

Reply:
Just to be precise, this fix is not fixing initial problem, when we investigated initial issue, we found two different problems.

I shared a behat test but I am not well experienced with it so I expect you correct me if I did anything wrong ^^'